core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -129,14 +134,15 @@ public T acquire(Integer time, TimeUnit unit) {
         mTakeLock.unlock();
       }
     } catch (InterruptedException e) {
-      throw new RuntimeException(e);
+      throw new IOException(e);
```

Comment:
DynamicResource throws IOException because of a different reason (a real IOException thrown when creating the resource).